xwiki-commons-core/xwiki-commons-job/xwiki-commons-job-default/src/main/java/org/xwiki/job/internal/DefaultJobStatusStore.java

```diff
@@ -378,7 +406,8 @@ private void saveJobStatus(JobStatus status)
             this.writeLock.lock();
 
             try {
-                File statusFile = getJobFolder(status.getRequest().getId());
+                // FIXME: not clear if the job status should be moved here.
```

Comment:
@tmortagne Have you seen this FIXME?

Reply:
I indeed missed it. It's not clear to me why it's a `FIXME`, actually.

You are wondering if it's right to move the status just to immediately overwrite it ?

In quite a lot of use cases, that id will go through get getJobStatus (which does the move, from what I understand) before. So I don't see why those use cases where there is no getJobStatus before the save should have a different end result, if anything at least for consistency.

Reply:
Yes, okay, I've removed the FIXME. I agree, I just wasn't 100% sure.